cypress/integration/login_test.js

```diff
@@ -101,4 +103,65 @@ describe('Login tests', () => {
       ]);
     });
   });
+
+  context('login form on multiple auth methods', () => {
+    beforeEach(() => {
+      cy.fixture('config/multiple_auth').as('config');
+    });
+
+    it('allows switching from OAuth to token login', function () {
+      const oauthLoginConfig = {
+        ...this.config,
+        default_auth: 'oauth',
+      };
+      cy.dynamicIntercept('getUserConfig', 'GET', '/user-config/config.json', oauthLoginConfig);
+      cy.visit('/login');
+      cy.contains('.btn-link', 'Switch to token login');
+    });
+
+    it('allows switching from token to OAuth login', function () {
+      const tokenLoginConfig = {
+        ...this.config,
+        default_auth: 'token',
+      };
+      cy.dynamicIntercept('getUserConfig', 'GET', '/user-config/config.json', tokenLoginConfig);
+      cy.visit('/login');
+      cy.contains('.btn-link', 'Switch to OAuth login');
+    });
+  });
+
+  context('OAuth login', () => {
+    beforeEach(() => {
+      cy.fixture('config/oauth').as('config');
+    });
+
+    it('does not show OAuth URL field when already configured', function () {
+      const config = _.cloneDeep(this.config);
+      config.auth[0].oauth_api_url = 'http://www.oauth.example.com';
+      cy.dynamicIntercept('getUserConfig', 'GET', '/user-config/config.json', config);
+      cy.visit('/login');
+
+      cy.get('input[id=astarteRealm]').should('be.empty');
+      cy.get('input[id=oauthProviderUrl]').should('not.exist');
+    });
+
+    /*
+      Cypress cannot intercept redirects to external URLs.
+      reenable then cypress add support to multi-domain testing
```

Comment:
`reenable then cypress add support to multi-domain testing` -> `TODO: reenable when cypress adds support to multi-domain testing`